From 2a47e333552d1186224d7ddc7a2be8a069fc4f45 Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Wed, 19 Oct 2016 12:32:10 -0500
Subject: [PATCH 1231/5242] MLK-13361-3 arm: imx6q: busfreq: restore mmdc
 timing settings for 100MHz

commit  22403643a2067f09cd2b1b62e717c78e3e772636 from
https://source.codeaurora.org/external/imx/linux-imx.git

The timing settings for 100MHz are almost the same as the ones for
400MHz except for the MMDCx_MISC[RALAT] parameter which needs to be
set to 2 cycles.

For the 100MHz case the restoration of the mmdc setting should be performed
in 2 steps: restore the mmdc setting and then overwrite the RALAT setting
for 2 cycles.

A decision code within the "mmdc_clk_lower_equal_100MHz" macro is added
to go to the "equal to 100MHz" or to the "lower to 100MHz" case

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
Signed-off-by: Alejandro Lozano <alejandro.lozano@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/lpddr2_freq_imx6q.S |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/mach-imx/lpddr2_freq_imx6q.S b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
index e14920d..593363a 100644
--- a/arch/arm/mach-imx/lpddr2_freq_imx6q.S
+++ b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
@@ -227,7 +227,17 @@ skip_below_100Mhz_ch1_timings:
 
 	.macro	mmdc_clk_lower_equal_100MHz
 
+	ldr     r10, =100000000
+	cmp     r0, r10
+	beq	set_timmings_100MHz
 	set_timings_below_100MHz_operation
+	b	common_to_lower_equal_100MHz
+
+set_timmings_100MHz:
+	restore_mmdc_settings_info
+	set_mmdc_misc_ralat_2_cycles
+
+common_to_lower_equal_100MHz:
 	/*
 	 * Prior to reducing the DDR frequency (at 528/400 MHz),
 	 * read the Measure unit count bits (MU_UNIT_DEL_NUM)
-- 
1.7.9.5

